File: VT09/VT09.cpp
#include<bits/stdc++.h>
using namespace std;

const int MAXN = 1000000;
bool dp_array[MAXN + 10];

void sang_snt() {
    dp_array[0] = dp_array[1] = false;
    for (int index_i = 2; index_i <= sqrt(MAXN); index_i++) {
        if (dp_array[index_i]) {
            for (int index_j = index_i * index_i; index_j <= MAXN; index_j += index_i) {
                dp_array[index_j] = false;
            }
        }
    }
}

int main() {
    fill(dp_array, dp_array + MAXN + 10, true);
    sang_snt();
    int number_n;
    cin >> number_n;
    set<int> number_set;
    for (int i = 0; i < number_n; i++) {
        int x;
        cin >> x;
        number_set.insert(abs(x));
    }
    vector<int> number_array(number_set.begin(), number_set.end());
    sort(number_array.begin(), number_array.end());
    for (int item : number_array) {
        if (dp_array[item]) {
            cout << item << " ";
        }
    }
    cout << endl;
    return 0;
}

File: VT09/VT09.py
import math
MAXN = int(1e6)
dp_array = [True] * (MAXN + 10)

def sang_snt():
    global dp_array
    dp_array[0]=dp_array[1]=False
    for index_i in range(2,int(math.sqrt(MAXN))+1):
        if dp_array[index_i]:
            for index_j in range(int(pow(index_i,2)),int(MAXN)+1,index_i):
                dp_array[index_j] = False

sang_snt()
number_n = int(input())
number_array = list(set(map(int,input().split())))
number_array.sort()
for item in number_array:
    if dp_array[abs(item)]:
        print(abs(item), end=" ")